connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -1981,6 +1966,8 @@ private void reconfigureConnector(final String connName, final Callback<Void> cb
                         }
                     });
                 }
+            } else {
+                log.debug("Skipping reconfiguration of connector {} as generated configs appear unchanged", connName);
```

Comment:
Why not move this into `AbstractHerder::taskConfigsChanged` as well, so that it's picked up in both standalone and distributed mode?